### PR TITLE
feat: flight deck improvements

### LINKED ISSUE #29: CTRK TK labels wrongly positioned
OVHD 320-251N

FUEL panel of the A320NEO has to be the same as from an A321.

### LINKED ISSUE #30: Remove ENG N1 mode switches
Currently the OVHD ENG panel is from a IAE engine of a CEO airbus.

The NEO has no N1 Mode switches. Depending on eng version there has to be a dual cooling switch for the A320-271N.
The A320-251N has just nothing.

OVHD 320-271N

### LINKED ISSUE #346: No Portable Elec Device option
This is sort of silly, but at my airline many newer Airbus aircraft come with a "no electronic device" sign instead of a seatbelt sign. No smoking is included as a permanent placard. Unless the MSFS A320neo is a retrofitted 320, I expected to see this difference when I first launched it.

The NO SMOKING sign is a "NO PORTABLE ELEC DEVICE" sign (see pic if I can attach it). Turning this sign off, which is usually done at above 10,000 ft, also extinguishes the lighted exit signs in the cabin for passenger comfort (nobody is evacuating at cruise altitude).

no_ped

### LINKED ISSUE #575: [BUG] Things light up that shouldn't
**Mod Version**

master

**Describe the bug**

There are things in the plane that should not light up as part of the integrated lights, but they do in the game.

- [x] parking brake #566
- [x] eng starter switches #566
- [ ] gravity gear extn (texture?)
- [ ] chronometer texts (texture?)
- [ ] ovhd circuits (texture?)

**References**

IRL photos:
chronometer texts like RST shouldn't light up ever:
image

**Additional context**

### LINKED ISSUE #604: [REQUEST] Support making 3D model changes
Master tracking issue for the glTF importer/exporter effort.

The main goal is to be able to import the plane model into blender, make modifications, and export back to glTF + bin file, and have it work in the mod.

1st milestone to work towards, is being able to import the plane into blender, and then export, without any modifications, and have the resulting gltf + bin match the vanilla ones, that way we know the tools are working.

# High Level TODO

- [ ]  fork official gltf io
    - [ ]  get mesh to show up in game
        - [ ]  add asobo mesh optimization extension
        - [ ]  look into other reasons why the mesh wouldnt display ingame
- [ ]  exporter
- [ ]  importer
- [ ]  import, export, then import it into blender AGAIN

I'm currently working on my fork of the official blender gltf2 importer/exporter: https://github.com/AdenFlorian/glTF-Blender-IO-a32nx. Importer mostly works, a couple skinned mesh positioning issues. Exporter needs a lot of work.

[https://www.khronos.org/files/gltf20-reference-guide.pdf](https://www.khronos.org/files/gltf20-reference-guide.pdf)

[https://docs.blender.org/api/current/info_overview.html](https://docs.blender.org/api/current/info_overview.html)

### LINKED ISSUE #849: [BUG] Fire Extinguisher bug
**Mod Version**

dev version, newest msfs version

**Describe the bug**

The fire button illumination looks awful

**To Reproduce**

1.Pop the fire extinguisher
2.Use the test button
3.see how awful it looks

**References**

Discord username (if different from GitHub): LorDawid#6403

### LINKED ISSUE #908: [REQUEST] SIDESTICK PRIORITY FIX
**Is your feature request related to a problem? Please describe.**

**Describe the feature you'd like to see implemented**

The sidestick priority red arrows are missing and sides priorities are wrong. Also if it can be functional.

176991B1-DD3E-4DC5-A4B0-0973C3588301

**References**

54FB42B8-7F9A-4FAA-9959-9F7CB43B94C6

**Additional context**

Discord username (if different from GitHub):

### LINKED ISSUE #1060: [REQUEST] Visual enhancement of ELEC overhead panel
**Is your feature request related to a problem? Please describe.**

Suggested visual enhancement of the overhead electrical panel to be more accurate to the real A320

**Describe the feature you'd like to see implemented**

On the "ELEC" section of the overhead panel, the "AC ESS FEED" button should display "ALTN" when activated, and not "OFF" as it currently does. Moreover, the schematic green line should be connected to the "AC BUS 2" inscription to the right of the button, to accurately represent its functionnality (see links and picture below for pictures of how it looks on the real thing).

**References**

http://www.a320dp.com/A320_DP/electrical/sys-3.2.5.html
https://www.youtube.com/watch?v=bPr3FUSIld4&list=PLdOwRGupI0JerxH2yupf782i4iO9Zf-vG&index=23

Overhead_panel_of_A320_aircraft_at_parking_stand

### LINKED ISSUE #1801: [BUG] Cockpit seats do not have their bottom textures
**Mod Version**

DEV Version from installer

**Describe the bug**

Chairs dont have their bottom textures

**To Reproduce**

1. Look at the bottom of chair

**Expected behavior**

There is a texture

**Actual behavior**

You can see-through the bottom of seats

**References**

**Additional context**

Desktop Screenshot 2020 11 08 - 12 58 27 59

Discord username (if different from GitHub):
LorDawid#6403

### LINKED ISSUE #2795: [REQUEST] Glareshield angle correction
**Is your feature request related to a problem? : No

**Describe the feature you'd like to see implemented**

We agree, the glasshield of the 320 from Asobo is disproportionate and gives the impression that the cockpit is more compact but also that the rendering is very ugly. The cockpit of an A320 Neo is identical to that of a 320 Classic so this rounded elevation of the glasshield is not necessary. We can now remedy this as we can export and re-import a 3D model, I propose that we correct the cockpit to show the beauty of the A320.

**References**

The green line represents the perfect symmetry while the red is what we have now.
Microsoft Flight Simulator Screenshot 2020 12 29 - 21 05 51 01
c49becb796ceb168b5086c92459755dd
Microsoft Flight Simulator Screenshot 2020 12 29 - 21 06 01 02
Below you can see the fairly large gap
cockpit2-3-01

**Additional context**
I hope that this will be taken into account but I also see the difficulty for the correction as it is a large part of the cockpit that will have to be modified.

Discord username : Sawada_Sensei

### LINKED ISSUE #3531: APU EXTING spelling error
**Mod Version**

d9a7c3d22521105ab367cbe8b9aeb80ee6b0ccef

**Describe the bug**

The APU panel at the far top and back on the OHP is labeled as `AUTO EXITING` instead of `AUTO EXTING` (as in "extinguish")

**To Reproduce**

1. Load up the aircraft
2. Look at the OHP
3. Look at the reference

**Expected behavior**

image

**Actual behavior**

image

**References**

**Additional context**

Was this working before/when did the issue start occurring?

Is this a problem in the vanilla unmodded game?

Discord username (if different from GitHub):

### LINKED ISSUE #3608: [BUG] Light coming through dash
lighting

**Mod Version**
"master" 

**Describe the bug**
Light coming through solid objects such as through the front dashboard 

**To Reproduce**
I connect ground power and the caution light that is projected from the vehicle comes through

**Expected behavior**
Light not to come through :)

**Additional context**

Is this a problem in the vanilla unmodded game? yes

### LINKED ISSUE #4207: Cabin window cubemap is facing the wrong way
**Mod Version**
{
  "built": "2021-03-30T22:44:03+00:00",
  "ref": "autopilot",
  "sha": "c7e6cbfe8a6e5d020d499a39aab1480a4ad6a943",
  "actor": "aguther",
  "event_name": "manual"
}

**Describe the bug**
Not really something serious but the cabin seats are placed backwards, facing the tail of the plane. 

**To Reproduce**
1. Use the drone in showcase camera
2. Move the drone close to a window
3. Adjust the angle if necessary to get a better view of the interior

**Expected behavior**
Seats should be facing the cockpit.

**Actual behavior**
Seats are facing the tail of the aircraft.

**References**
I don't think this is really needed.

**Additional context**
https://prnt.sc/110cyy2

Was this working before/when did the issue start occurring?
Yes, it was working before. I can't really pinpoint when this issue occured but a couple of weeks ago the cabin layout was correct.

Is this a problem in the vanilla unmodded game?
No idea. I don't use the vanilla A320.

Discord username (if different from GitHub):
Codehack#6987

Seat Orientation

### LINKED ISSUE #4584: Movable armrest for better view on center pedestal
**Is your feature request related to a problem? Please describe.**
Some buttons are not accessable from the pilot's view as the inboard armrest blocks the view.

**Describe the feature you'd like to see implemented**
A clickspit to move the armrest up and down would be nice to get a better view from the pilot's seat without moving around the camera in the cockpit.

**References**
Some other flight sim aircraft have this feature. The inibuild a300 for xplane for example.

**Additional context**
N/A

Discord username (if different from GitHub):

### LINKED ISSUE #4595: Textures for VU120-VU122 circuit breaker panels at cockpit back wall
**Aircraft Version**

{
  "built": "2021-04-23T22:37:56+00:00",
  "ref": "experimental",
  "sha": "c43d0267084cae66e5cb3950c1c76e9b78bf43d0",
  "actor": "aguther",
  "event_name": "manual"
}

**Describe the bug**
The textures/ lettering for the aft wall circurit breakers is missing. The OVHD panel is way better.

**To Reproduce**
Turn around and look back.

**Expected behavior**
Same texture quality as on the OVHD panel.

**Actual behavior**
Asobo state of work.

**References**
N/A

**Additional context**
N/A

Was this working before/when did the issue start occurring?
N/A

Is this a problem in the vanilla unmodded game?
YES

Discord username (if different from GitHub):

### LINKED ISSUE #5139: Emissive for LDG GEAR down
### Aircraft Version

Development

### Description

Emissive (red) for LDG GEAR down is missing and should be added.

Not many details here, as this issue is made to split up the remaining items in #528.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

SjotgunSjonnie#9623

### LINKED ISSUE #5142: RST, CHR, DATE, GPS/INT, and RUN/STP/RST shouldn't light up
### Aircraft Version

Development

### Description

The CLOCK's RST, CHR, DATE, GPS/INT, and RUN/STP/RST shouldn't light up.

This issue is taken from #956 (which will be closed).

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

SjotgunSjonnie#9623

### LINKED ISSUE #5234: Fix LDG ELEV legend
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-06-29T06:59:57+00:00",
  "ref": "HEAD",
  "sha": "4b526b9eaf3900b9949b1b08024135c467390478",
  "actor": "BBK <22713769+BlueberryKing@users.noreply.github.com>",
  "event_name": "manual"
}
```

### Describe the bug

Numbers on the LDG ELEV AUTO knob legend is cut off.

ldg-elev-auto

### Expected behavior

Correctly aligned legends.

### Steps to reproduce

Spawn.
Look at OVHD.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

bouveng

### LINKED ISSUE #5536: Missing dot on fire panel test buttons
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-08-04T18:35:13+00:00",
  "ref": "HEAD",
  "sha": "fe97ce409e1f9d55b11185de38ce069915934bc9",
  "actor": "Erick Torres <30361843+ErickSharp@users.noreply.github.com>",
  "event_name": "manual"
}
```

### Describe the bug

The white dot's, introduced by #5111 have been removed as a result of #5497.
missing-dots

### Expected behavior

The aforementioned buttons should have a white dot.

### Steps to reproduce

1. Spawn
2. Observe fire panel on the OVHD

### References (optional)

A320 OVHD
121844937-05b11300-cce5-11eb-831e-38e5594864f9
A32NX before #5497 
121844665-99ceaa80-cce4-11eb-9239-7f43e2b3830a

### Additional info (optional)

_No response_

### Discord Username (optional)

bouveng

### LINKED ISSUE #5607: Small texture bug on the Airbus overhead panel around the air conditioning exhaust
### Aircraft Version

Development

### Build info

```json
Newest Development build (but the texture bug is also present in the stable and the Experimental version.)
```

### Describe the bug

Hi and congratulations for this absolute impressive and beautyful Airbus!
In my opinion it is almost perfect and everything works fine, but I have found a little texture bug on the overhead panel on one of the air condition exhausts.
If you could take care of this texture bug the interior of the Airbus would be absolutely perfect :)
image
image

Thanks for contributing such a wonderful fantastic plane to this flight sim!

### Expected behavior

Texture bug on the overhead panel gone.

### Steps to reproduce

1. ) Sitting inside the Airbus
2.) looking at the overhead panel

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

Murmeltier Deluxe#9219

### LINKED ISSUE #5613: Screws in the remodelled glareshield/AP Section are not realistic anymore
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-08-10T08:14:38+00:00",
  "ref": "master",
  "sha": "d30667bee2a5571ee055371021bee4a7da30eefd",
  "actor": "davidwalschots",
  "event_name": "manual"
}
```

### Describe the bug

The screws in the new remodelled glareshield/AP section are not realistic anymore if you compare with the real A320 cockpit.

The screws are sticking out of the panel too much.

FBW Cockpit:

2021-08-10 17_26_50-Microsoft Flight Simulator - 1 18 15 0

### Expected behavior

Airbus is not using screws that are sticking out of the panels that much. All screws should be countersunk screws to avoid sharp edges and possible risk of injury. Also the color of the real screws is a little darker. They are probably anodized.

Real Airbus Cockpit:

2021-08-10 17_34_42-A320 - 360 cockpit view _ Virtual tour generated by Panotour

### Steps to reproduce

Start the FBW mod and look at the cockpit section that shows the glareshield and the autopilot.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### LINKED ISSUE #6153: OVHD SVCE INT OVRD spelling error
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-11-07T13:22:52+00:00",
  "ref": "HEAD",
  "sha": "812680b8be6a214b60e2a30fcd5380864e36b707",
  "actor": "Igor Milovidov <igor8518@mail.ru>",
  "event_name": "manual"
}
```

### Describe the bug

Spelling error on the **OVHD SVCE INT OVRD** button.
It says `SVGE` instead of `SVCE` (service)

svce-int-ovrd
 

### Expected behavior

Button should say `SVCE` instead of `SVGE`.

### Steps to reproduce

1. Spawn
2. Observe upper OVHD

### References (optional)

svce-reference

### Additional info (optional)

_No response_

### Discord Username (optional)

bouveng

### LINKED ISSUE #6251: Missing Caution Decals
### Aircraft Version

Development

### Description

I recently notice while watching a few yt videos that the aircraft is missing some warning labels at first I thought this was only for that certain airlines livery but quickly realized it wasn't as I checked many other walk around videos.

Below I've provided how things currently look like in the sim.

image
image
image
image

### References (optional)

image
image
image
image
image
video screenshots were taken from:
https://www.youtube.com/watch?v=LNPrGbdWIbU
more examples of those tag
https://www.youtube.com/watch?v=Tdf_Ac7_gwA
https://www.youtube.com/watch?v=2B7Ku9ufxTk
https://www.youtube.com/watch?v=5kxH_li4LIc

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### LINKED ISSUE #7622: ND Terrain INOP sticker still there, but Terrain ND button working
### Aircraft Version

Development

### Build info

```json
{
    "built": "2022-11-27T21:40:05+00:00",
    "ref": "refs/heads/master",
    "sha": "5909ecf2ff51aabd46e8c034f71bd33ce5184fd9",
    "actor": "frankkopp",
    "event_name": "manual",
    "pretty_release_name": "master:5909ecf2"
}
```

### Describe the bug

ND Terrain INOP sticker still there, but Terrain ND button working

### Expected behavior

No INOP sticker

### Steps to reproduce

Use FBW development version

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #29 
Fixes #30
Fixes #346
Fixes #575
Fixes #604
Fixes #849
Fixes #908
Fixes #1060
Most of #1410
Fixes #1801
Fixes #2795
Fixes #3531 
Fixes #3608
Fixes #4207
Fixes #4584
Fixes #4595
Fixes #5139
Fixes #5142
Fixes #5139
Fixes #5234
Fixes #5536
Fixes #5607
Fixes #5613
Fixes #6153 
Fixes #6251 (think that was already fixed by grinde)
Adds the button for #6924
Fixes #7622

## Summary of Changes

1. [EFB] Added new F/O camera view, and adjusted captain view to suit new flyPad orientation - @tracernz (Mike)
1. [OVHD] Fire button LEDs now split by power supply - @tracernz (Mike), @Repsol2k
1. [OVHD] No PED Label instead of NO SMOKING when selected - @tracernz (Mike)
1. [FWC/PANEL] GEAR NOT DOWN warnings added including red arrow on panel - @tracernz (Mike)
1. [MODEL] Fixed some glaring issues with the flight deck model - @MoreRightRudder, @Repsol2k, @tracernz
1. [MODEL] Replaced overhead decals - @Repsol2k
1. [MODEL] General improvements to 3d model and textures - @MoreRightRudder, @Repsol2k, @tracernz
1. [MODEL] Folding armrests - @Repsol2k 

Completely new and accurate flight deck seats! 😍 

Huge thanks to Repsol and Mike!!

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Testing already done. Big thanks to Alepouna, Alex R, danbw9, and Shomas!

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/systems/instruments/src/Common/EWDMessages.tsx [hunk 1]
```diff
@@ -347,6 +347,8 @@ const EWDMessages = {
     '320005001': '\x1b<2m\x1b4mCONFIG\x1bm PARK BRK ON',
     '320006001': '\x1b<4m\x1b4mBRAKES\x1bm A/SKID N/WS FAULT',
     '320006002': '\x1b<5m MAX BRK PR......1000 PSI',
+    '320015001': '\x1b<2m\x1b4mL/G\x1bm GEAR NOT DOWN',
+    '320015501': '\x1b<2m\x1b4mL/G\x1bm GEAR NOT DOWN',
```
Comment (Saschl) [inline fbw-a32nx/src/systems/instruments/src/Common/EWDMessages.tsx:351]: Is this expected to be identical?

Comment (tracernz) [reply fbw-a32nx/src/systems/instruments/src/Common/EWDMessages.tsx:351]: Yes, that is intentional. Good spotting though 😁 .

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FCU/A320_Neo_FCU.js [hunk 2]
```diff
@@ -1337,26 +1409,446 @@ class A320_Neo_FCU_Pressure extends A320_Neo_FCU_Component {
...
+        if (this.pressure1) {
+            this.pressure1.reboot();
+        }
+        if (this.pressure2) {
+            this.pressure2.reboot();
         }
     }
 }
 
+// Based on ftp://climate1.gsfc.nasa.gov/wiscombe/Solar_Rad/SunAngles/sunae.f
```
Comment (lukecologne) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FCU/A320_Neo_FCU.js:1443]: Does the FCU have automatic brightness adjustment IRL?

**Review by lukecologne — APPROVED** [review]
XML looks good to me, not sure about the FCU auto-brightness, needs confirmation but nothing major. The rest of the FCU is fine as well.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/cameras.cfg [hunk 3]
```diff
@@ -1701,7 +1701,7 @@ PanPbhReturn = 0
 Track = "None"
 ShowAxis = "YES"
 AllowZoom = 1
-InitialZoom = 0.14
+InitialZoom=0.113515
```
Comment (BlueberryKing) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/cameras.cfg:1704]: Just formatting
```suggestion
InitialZoom = 0.113515
```

File: fbw-a32nx/src/behavior/src/A32NX_Interior_RMP.xml [hunk 4]
```diff
@@ -218,52 +226,51 @@
...
         <UseTemplate Name="FBW_Anim_Interactions">
             <ANIM_TYPE>BUTTON</ANIM_TYPE>
             <ANIM_TEMPLATE>FBW_RMP_MODE_BUTTON_AND_LED</ANIM_TEMPLATE>
-            <BASE_NAME>MLS</BASE_NAME>
+            <BASE_NAME>GLS</BASE_NAME>
             <MODE_ID>8</MODE_ID>
         </UseTemplate>
 
-        <!-- Inop ADF mode button and led -->
+        <!-- GLS mode button and led -->
```
Comment (BlueberryKing) [inline fbw-a32nx/src/behavior/src/A32NX_Interior_RMP.xml:261]: I think there's a typo here
```suggestion
        <!-- MLS mode button and led -->
```

**Comment by mjuhe** [comment]
Awesome work. I can't add comments to unchanged files, but I believe for the landing elevation knob to work, you will need to add these changes to the \fbw-a32nx\src\wasm\systems\a320_systems\src\air_conditioning.rs file:

Line 891:
`ldg_elev_knob: ValueKnob::new_with_value(context, "PRESS_LDG_ELEV", -4000.),`

Line 938:
`self.ldg_elev_knob.value() < -(2000. + margin)`

Line 2248:
`test_bed = test_bed.command_ldg_elev_knob_value(-4000.).and_run();`

Otherwise I can open a separate PR for it once this one is merged.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FCU/A320_Neo_FCU.js (hunk 2 above)
Comment (tracernz) [reply fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FCU/A320_Neo_FCU.js:1443]: No, but the ambient lighting has a great effect on it's appearance, which is what we adjust with this :D.

**Comment by auroraisluna** [comment]
QA Report

Discord: alepouna
Object of testing:  #8334
Tier of Testing : 1
Date: 21/12/2023

Testing Process:
- Inspected the model, try all switches, various camera angles. 
- Performed a quick flight with on a yesterday build that had some decal bugs (which are fixed in the latest version) so could consider this Tier 2 as well. 
- Tested a variety of liveries.  

Testing Results:
Passed

Notes: 
Great job! This is 🔥🔥

**Pull request merged**